R/edit.R

```diff
@@ -15,65 +17,100 @@ NULL
 #' @export
 #' @rdname edit
 edit_r_profile <- function(scope = c("user", "project")) {
-  edit_file(scope_dir(scope), ".Rprofile")
+  file <- edit_file(scoped_path(scope, ".Rprofile"))
   todo("Restart R for changes to take effect")
-  invisible()
+  invisible(file)
 }
 
 #' @export
 #' @rdname edit
 edit_r_environ <- function(scope = c("user", "project")) {
-  edit_file(scope_dir(scope), ".Renviron")
+  file <- edit_file(scoped_path(scope, ".Renviron"))
   todo("Restart R for changes to take effect")
-  invisible()
+  invisible(file)
 }
 
 #' @export
 #' @rdname edit
 edit_r_makevars <- function(scope = c("user", "project")) {
-  edit_file(scope_dir(scope), ".R/Makevars")
+  file <- edit_file(scoped_path(scope, ".R", "Makevars"))
   todo("Restart R for changes to take effect")
-  invisible()
+  invisible(file)
 }
 
 #' @export
 #' @rdname edit
-edit_git_config <- function(scope = c("user", "project")) {
-  scope <- match.arg(scope)
-  path <- switch(scope, user = ".gitconfig", project = ".git/config")
-  edit_file(git_scope_dir(scope), path = path)
-  invisible()
+#' @param type Snippet type. One of "R", "markdown", "C_Cpp", "Tex",
+#'   "Javascript", "HTML", "SQL"
+edit_rstudio_snippets <- function(type = "R") {
+  file <- scoped_path(
+    "user",
+    ".R", "snippets", paste0(tolower(type), ".snippets")
+  )
+  invisible(edit_file(file))
 }
 
+
+# git files are special ----
+
 #' @export
 #' @rdname edit
-edit_git_ignore <- function(scope = c("user", "project")) {
-  ## TODO(jennybc) https://github.com/r-lib/usethis/issues/182
-  edit_file(git_scope_dir(scope), ".gitignore")
-  invisible()
+edit_git_config <- function(scope = c("user", "project")) {
+  path <- switch(
+    scope,
+    user = ".gitconfig",
+    project = file.path(".git", "config")
+  )
+  invisible(edit_file(scoped_git_path(scope, path)))
 }
 
 #' @export
 #' @rdname edit
-#' @param type Snippet type. One of "R", "markdown", "C_Cpp", "Tex",
-#'   "Javascript", "HTML", "SQL"
-edit_rstudio_snippets <- function(type = "R") {
-  edit_file("~", paste0(".R/snippets/", tolower(type), ".snippets"))
-  invisible()
+edit_git_ignore <- function(scope = c("user", "project")) {
+  file <- scoped_git_path(scope, ".gitignore")
+  if (scope == "user" && !file.exists(file)) {
+    todo("Tell git about this new global gitignore file for changes to take effect!")
```

Comment:
I did not know this!

Reply:
Maybe I should just call `git2r::git2r::config()` for them in this case and be done with it?

Reply:
Yeah, I think that's reasonable